app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemViewHolder.kt

```diff
@@ -63,12 +71,32 @@ class TopSiteItemViewHolder(
             }
             val menu = topSiteMenu.menuBuilder.build(view.context).show(anchor = it)
 
-            it.setOnTouchListener @SuppressLint("ClickableViewAccessibility") { v, event ->
+            it.setOnTouchListener { v, event ->
                 onTouchEvent(v, event, menu)
             }
 
             true
         }
+
+        store.flowScoped(viewLifecycleOwner) { flow ->
+            flow.map { state -> state.wallpaperState.currentWallpaper }
+                .ifChanged()
+                .collect { currentWallpaper ->
+                    val isDarkMode = view.resources.configuration.uiMode and
```

Comment:
2 ideas here - I am wondering if we could cheat by using `isSystemInDarkTheme()`

Alternatively, I saw we have this also defined as an extension method in https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt#366. Wondering if we could refactor that `ext/Context` to avoid the duplication of logic?

Reply:
I'll look into this

Reply:
Opened https://github.com/mozilla-mobile/fenix/issues/27456

I'll land a fix for ^ to integrate here